File: trunk/generic/client/gwt/src/main/java/org/cggh/chassis/generic/client/gwt/configuration/client/Configuration.java
/**
 * 
 */
package org.cggh.chassis.generic.client.gwt.configuration.client;


import com.google.gwt.core.client.JsArray;

/**
 * @author aliman
 *
 */
public class Configuration {

	public static native String getUserDetailsServiceEndpointURL() /*-{
		return $wnd.CHASSIS.userDetailsServiceEndpointURL;
	}-*/;
	
	public static native String getUserChassisRolesPrefix() /*-{
		return $wnd.CHASSIS.userChassisRolesPrefix;
	}-*/;
	
	public static native JsChassisRole getChassisRoleAdministrator() /*-{
		return $wnd.CHASSIS.chassisRoleAdministrator;
	}-*/;

	public static native JsChassisRole getChassisRoleCoordinator() /*-{
		return $wnd.CHASSIS.chassisRoleCoordinator;
	}-*/;
	
	public static native JsChassisRole getChassisRoleCurator() /*-{
		return $wnd.CHASSIS.chassisRoleCurator;
	}-*/;
	
	public static native JsChassisRole getChassisRoleGatekeeper() /*-{
		return $wnd.CHASSIS.chassisRoleGatekeeper;
	}-*/;
	
	public static native JsChassisRole getChassisRoleSubmitter() /*-{
		return $wnd.CHASSIS.chassisRoleSubmitter;
	}-*/;
	
	public static native JsChassisRole getChassisRoleUser() /*-{
		return $wnd.CHASSIS.chassisRoleUser;
	}-*/;
	
	public static native String getStudyFeedURL() /*-{
		return $wnd.CHASSIS.studyFeedURL;
	}-*/;

	public static native String getSubmissionFeedURL() /*-{
		return $wnd.CHASSIS.submissionFeedURL;
	}-*/;

	public static native String getDataFileFeedURL() /*-{
		return $wnd.CHASSIS.dataFileFeedURL;
	}-*/;


	public static native String getSandboxFeedURL() /*-{
		return $wnd.CHASSIS.sandboxFeedURL;
	}-*/;

	public static native String getDataFileUploadServiceURL() /*-{
		return $wnd.CHASSIS.dataFileUploadServiceURL;
	}-*/;

	/**
	 * @return
	 */
	public static native String getSubmissionQueryServiceURL()  /*-{
		return $wnd.CHASSIS.submissionQueryServiceURL;
	}-*/;

	public static native JsArray<Module> getModules() /*-{
		return $wnd.CHASSIS.modules;
	}-*/;

	/**
	 * @return
	 */
	public static native String getStudyQueryServiceURL() /*-{
		return $wnd.CHASSIS.studyQueryServiceURL;
	}-*/;
	
	public static native String getDataFileQueryServiceURL() /*-{
		return $wnd.CHASSIS.dataFileQueryServiceURL;
	}-*/;
	
	public static native String getStudyQuestionnaireURL() /*-{
		return $wnd.CHASSIS.studyQuestionnaireURL;
	}-*/;
}

File: trunk/generic/client/gwt/war/configuration/configuration.js
if (typeof CHASSIS == "undefined" || !CHASSIS) {
	CHASSIS = new Object();
}

// configure service URLs

CHASSIS.client = "/chassis-generic-client-gwt";
CHASSIS.userService = "/chassis-generic-service-user";
CHASSIS.existService = "/chassis-generic-service-exist";
CHASSIS.uploadService = "/chassis-generic-service-upload";

CHASSIS.userDetailsServiceEndpointURL = CHASSIS.userService + "/gwtrpc/userdetails";

CHASSIS.studyFeedURL = CHASSIS.existService + "/atom/edit/studies";
CHASSIS.studyQueryServiceURL = CHASSIS.existService + "/query/studies.xql";

CHASSIS.studyQuestionnaireURL = CHASSIS.client + "/questionnaire/study-questionnaire.xml";

CHASSIS.submissionFeedURL = CHASSIS.existService + "/atom/edit/submissions";
CHASSIS.submissionQueryServiceURL = CHASSIS.existService + "/query/submissions.xql";

CHASSIS.dataFileFeedURL = CHASSIS.existService + "/atom/edit/datafiles";
CHASSIS.dataFileQueryServiceURL = CHASSIS.existService + "/query/datafiles.xql";

CHASSIS.mediaFeedURL = CHASSIS.existService + "/atom/edit/media";

CHASSIS.sandboxFeedURL = CHASSIS.existService + "/atom/edit/sandbox";

CHASSIS.newDataFileServiceURL = CHASSIS.uploadService + "/newdatafile";
CHASSIS.uploadDataFileRevisionServiceURL = CHASSIS.uploadService + "/revisedatafile";

// CHASSIS.dataFileUploadServiceURL = CHASSIS.uploadService + "/datafileupload";

// configure role name prefix

CHASSIS.userChassisRolesPrefix = "ROLE_CHASSIS_";

// configure permission suffix to chassis role mappings

CHASSIS.chassisRoleCoordinator = {
	"permissionSuffix": "COORDINATOR",
	"label": {
		"en": "Coordinator"
	}
};

CHASSIS.chassisRoleCurator = {
	"permissionSuffix": "CURATOR",
	"label": {
		"en": "Curator"
	}
};

CHASSIS.chassisRoleGatekeeper = {
	"permissionSuffix": "GATEKEEPER",
	"label": {
		"en": "Gate Keeper"
	}
};

CHASSIS.chassisRoleSubmitter = {
	"permissionSuffix": "SUBMITTER",
	"label": {
		"en": "Submitter"
	}
};

CHASSIS.chassisRoleUser = {
	"permissionSuffix": "USER",
	"label": {
		"en": "User"
	}
};

CHASSIS.chassisRoleAdministrator = {
	"permissionSuffix": "ADMINISTRATOR",
	"label": {
		"en": "Administrator"
	}
};

// configure modules
CHASSIS.modules = [
   {
	   "id":"moduleA",
	   "label": {
	   		"en": "Module A"
   		}
   },
   {
	   "id": "moduleB",
	   "label": {
	   		"en": "Module B"
   		}
   },
   {
	   "id": "moduleC",
	   "label": {
	   		"en": "Module C"
   		}
   }
];